tests/parametric/test_parametric_endpoints.py

```diff
@@ -0,0 +1,733 @@
+                s1.set_status("ERROR", "error message")
+                s1.end_span()
+
+        traces = test_agent.wait_for_num_traces(1)
+        span = find_only_span(traces)
+        assert span["error"] == 1
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_OtelSpan_Set_Name:
+    def test_otelspan_set_name(self, test_agent, test_library):
+        """
+        Validates that /trace/otel/set_name sets a name on a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        - name: str
+        Supported Return Values:
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_set_name") as s1:
+                s1.set_name("new_name")
+                s1.end_span()
+
+        traces = test_agent.wait_for_num_traces(1)
+        span = find_only_span(traces)
+        assert span["resource"] == "new_name"
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_OtelSpan_Events:
+    def test_add_event(self, test_agent, test_library):
+        """
+        Validates that /trace/otel/add_event adds an event to a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        - name: str
+        - timestamp (μs): Optional[int]
+        - attributes: Optional[Dict[str, str]]
+        Supported Return Values:
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_add_event") as s1:
+                s1.add_event("some_event", 1730393556000000, {"key": "value"})
+                s1.end_span()
+
+        traces = test_agent.wait_for_num_traces(1)
+        span = find_only_span(traces)
+        assert "events" in span["meta"]
+        events = json.loads(span["meta"]["events"])
+        assert len(events) == 1
+        assert events[0]["name"] == "some_event"
+        assert events[0]["time_unix_nano"] == 1730393556000000000
+        assert events[0]["attributes"]["key"] == "value"
+
+    @irrelevant(context.library == "golang", reason="OTEL does not expose an API for recording exceptions")
+    def test_record_exception(self, test_agent, test_library):
+        """
+        Validates that /trace/otel/record_exception adds an exception event to a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
+        - message: str
+        - attributes: str
+        Supported Return Values:
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_record_exception") as s1:
+                s1.record_exception("MyException Parametric tests rock", {"error.key": "value"})
+                s1.end_span()
+
+        traces = test_agent.wait_for_num_traces(1)
+        span = find_only_span(traces)
+        assert "events" in span["meta"]
+        events = json.loads(span["meta"]["events"])
+        assert len(events) == 1
+        assert events[0]["name"].lower() in ["exception", "error"]
```

Comment:
Can you also assert that the attribute is set correctly?

Reply:
Great call out! Adding this assertion surfaced an issue in the js parametric app: https://github.com/DataDog/system-tests/pull/3345/commits/5be26e7ca0c551ad782092693df0f15ec9931c2f.

The record exception endpoint ignores the attributes parameter: https://github.com/DataDog/system-tests/blob/278f73ae5111a4ec4f1313e66250d86c073714a2/utils/build/docker/nodejs/parametric/server.js#L323.